kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerLocator.java

```diff
@@ -15,19 +15,37 @@
 
 package org.kie.server.services.impl;
 
+import org.kie.server.api.KieServerConstants;
+import org.kie.server.services.impl.storage.cloud.KieServerStateCloudRepository;
+
 public class KieServerLocator {
 
-    private static final KieServerImpl INSTANCE = new KieServerImpl();
-    
+    private static final KieServerImpl INSTANCE;
+    private static String strategy = System.getProperty(KieServerConstants.KIE_SERVER_STARTUP_STRATEGY);
+
     static {
+        if (strategy == null || strategy.length() == 0) {
+            strategy = System.getenv("KIE_SERVER_STARTUP_STRATEGY");
+
+            if (strategy == null || strategy.length() == 0) {
+                strategy = ControllerBasedStartupStrategy.class.getSimpleName();
+            }
+
+            System.setProperty(KieServerConstants.KIE_SERVER_STARTUP_STRATEGY, strategy);
+        }
+
+        if ("OpenShiftStartupStrategy".equals(strategy) || "KubernetesStartupStrategy".equals(strategy)) {
```

Comment:
Maybe it would have sense to use service loader to load KieServerStateRepository, it would allow bigger flexibility while keeping basic Kie server functionality isolated from different implementations and their dependencies. What do you think about it?

Reply:
Noticed that there are already two different styles of loading dependencies. At StartupStrategyProvider class, service loader is used. 

https://github.com/rhtevan/droolsjbpm-integration/blob/abf0dcad8de3b03909649fd1f67af4e0b4e1c0e5/kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/StartupStrategyProvider.java#L31-L32

On the other hand, KieServerImpl class is rather taking a simple approach. 

https://github.com/rhtevan/droolsjbpm-integration/blob/abf0dcad8de3b03909649fd1f67af4e0b4e1c0e5/kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java#L101-L103

At the moment, we'd like to keep this re-engineering task focused, and introduce changes which are absolutely necessary. (KieServerImpl class is untouched.) Moreover, we may have already be experiencing thread safe issues on current StartupStrategyProvider implementation with ServiceLoader approach. Perhaps, an overhaul with CDI approach could be next. For now, KISS principle applies.